src/html-media-source.js

```diff
@@ -145,14 +155,43 @@ export default class HtmlMediaSource extends videojs.EventTarget {
       }
 
       buffer = new VirtualSourceBuffer(this, [avcCodec, mp4aCodec]);
-      this.sourceBuffers.push(buffer);
-      return buffer;
+    } else {
+      // delegate to the native implementation
+      buffer = this.mediaSource_.addSourceBuffer(type);
     }
 
-    // delegate to the native implementation
-    buffer = this.mediaSource_.addSourceBuffer(type);
+    setActiveSourceBuffers = () => {
+      this.updateActiveSourceBuffers_();
+      // TODO buffer.one causing call stack exceeded exception
+      buffer.off('updateend', setActiveSourceBuffers);
+    };
+    // for combined audio/video tracks, we can only determine if a source buffer is
+    // active after a completed update (once it has/doesn't have videoTracks)
+    buffer.on('updateend', setActiveSourceBuffers);
+
     this.sourceBuffers.push(buffer);
     return buffer;
   }
+
+  updateActiveSourceBuffers_() {
+    // Retain the reference but empty the array
+    this.activeSourceBuffers_.length = 0;
+
+    if (this.player_.audioTracks().some((audioTrack) => audioTrack.enabled)) {
+      // We are using an alternate audio track from the default. Since we currently only
+      // support a max of two source buffers, add all of the source buffers (in order).
+      this.sourceBuffers.forEach((sourceBuffer) => {
+        this.activeSourceBuffers_.push(sourceBuffer);
+      });
+    } else {
+      // We are using the combined audio/video stream, so only add the combined source
+      // buffer.
+      this.sourceBuffers.forEach((sourceBuffer) => {
+        if (sourceBuffer.videoTracks && sourceBuffer.videoTracks.length > 0) {
```

Comment:
We are just using videoTracks to see what source buffer represents video/combined, since we don't have a way of identifying it otherwise (I may be reading it wrong, but it seems that all source buffers are added with the mimeType of just 'video/mp2t' with no codecs specified, from https://github.com/videojs/videojs-contrib-hls/blob/my-rebase/src/segment-loader.js#L145 ).